service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/ObservationMapper.java

```diff
@@ -250,39 +286,134 @@ private String mapObservationToObservationStatement(MultiStatementObservationHol
+                    .append(prepareReferenceRangeToComment(referenceRange));
+            }
         }
 
+        Optional.of(interpretationTextAndComment)
+            .map(StringBuilder::toString)
+            .filter(StringUtils::isNotBlank)
+            .map(String::trim)
+            .map(textAndComment ->
+                mapObservationToNarrativeStatement(
+                    holder, textAndComment, prepareCommentType(observation).getCode())
+            ).ifPresent(narrativeStatementsBlock::append);
+
         if (!narrativeStatementsBlock.toString().isBlank()) {
             return Optional.of(narrativeStatementsBlock.toString());
         }
 
         return Optional.empty();
     }
 
+    private Optional<String> prepareNarrativeStatementForRelatedObservationComments(MultiStatementObservationHolder holder) {
+        Observation observation = holder.getObservation();
+
+        StringBuilder relatedObservationsComments = new StringBuilder();
+        if (observation.hasRelated()) {
+            List<Observation> validObservations = observation.getRelated()
+                .stream()
+                .filter(this::hasValidType)
+                .map(observationRelatedComponent -> observationRelatedComponent.getTarget().getResource())
+                .map(Observation.class::cast)
+                .filter(this::hasValidComment)
+                .collect(Collectors.toList());
+
+            if (!validObservations.isEmpty()) {
+                final IdMapper idMapper = messageContext.getIdMapper();
+
+                validObservations.forEach(validObservation -> {
+                    idMapper.markObservationAsMapped(validObservation.getIdElement());
+                    relatedObservationsComments.append(StringUtils.LF).append(validObservation.getComment());
+                });
+            }
+
+            StringBuilder narrativeStatementsBlock = new StringBuilder();
+
+            Optional.of(relatedObservationsComments)
+                .map(StringBuilder::toString)
+                .filter(StringUtils::isNotBlank)
+                .map(textAndComment -> mapObservationToNarrativeStatement(
+                        holder, textAndComment, CommentType.USER_COMMENT.getCode())
+                ).ifPresent(narrativeStatementsBlock::append);
+
+            if (!narrativeStatementsBlock.toString().isBlank()) {
+                return Optional.of(narrativeStatementsBlock.toString());
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    private CommentType prepareCommentType(Observation observation) {
+        return hasCode(observation.getCode(), List.of(COMMENT_NOTE_CODE))
+            ? CommentType.USER_COMMENT : CommentType.AGGREGATE_COMMENT_SET;
+    }
+
+    private String prepareReferenceRangeValues(SimpleQuantity simpleQuantity) {
+        return Stream.of(
+            Optional.ofNullable(simpleQuantity.getValue()),
+            Optional.ofNullable(simpleQuantity.getUnit())
```

Comment:
Maybe if those optionals were indented more it would be more readable?

Reply:
Right, thanks :) It's done now